packages/credential-eip712/src/types/ICredentialEIP712.ts

```diff
@@ -0,0 +1,186 @@
+   * @param args - Arguments necessary to create the Presentation.
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   *
+   * @returns - a promise that resolves to the {@link @veramo/core#VerifiablePresentation} that was requested or rejects with an error
+   * if there was a problem with the input or while getting the key to sign
+   *
+   * @remarks Please see {@link https://www.w3.org/TR/vc-data-model/#presentations | Verifiable Presentation data model }
+   */
+  createVerifiablePresentationEIP712(
+    args: ICreateVerifiablePresentationEIP712Args,
+    context: IRequiredContext,
+  ): Promise<VerifiablePresentation>
+
+  /**
+   * Verifies a Verifiable Presentation EIP712 Format.
+   *
+   * @param args - Arguments necessary to verify a VerifiableCredential
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   *
+   * @returns - a promise that resolves to the boolean true on successful verification or rejects on error
+   *
+   * @remarks Please see {@link https://www.w3.org/TR/vc-data-model/#presentations | Verifiable Credential data model}
+   */
+  verifyPresentationEIP712(
+    args: IVerifyPresentationEIP712Args,
+    context: IRequiredContext
+  ): Promise<boolean>  
+}
+
+/**
+ * Encapsulates the parameters required to create a
+ * {@link https://www.w3.org/TR/vc-data-model/#credentials | W3C Verifiable Credential}
+ *
+ * @beta This API is likely to change without a BREAKING CHANGE notice
+ */
+export interface ICreateVerifiableCredentialEIP712Args {
+  /**
+   * The json payload of the Credential according to the
+   * {@link https://www.w3.org/TR/vc-data-model/#credentials | canonical model}
+   *
+   * The signer of the Credential is chosen based on the `issuer.id` property
+   * of the `credential`
+   *
+   * `@context`, 'type' and 'issuanceDate' will be added automatically if omitted
+   */
+  credential: CredentialPayload
+
+  /**
+   * Specific key to use for signing
+   */
+  keyRef?: string
+}
+
+export interface ICreateVerifiablePresentationEIP712Args {
+  /**
+   * The json payload of the Presentation according to the
+   * {@link https://www.w3.org/TR/vc-data-model/#presentations | canonical model}.
+   *
+   * The signer of the Presentation is chosen based on the `holder` property
+   * of the `presentation`
+   *
+   * '@context', 'type' and 'issuanceDate' will be added automatically if omitted
+   */
+  presentation: PresentationPayload
+
+  /**
+   * [Optional] The ID of the key that should sign this presentation.
+   * If this is not specified, the first matching key will be used.
+   */
+  keyRef?: string  
+}
+
+/**
+ * Encapsulates the parameters required to verify a
+ * {@link https://www.w3.org/TR/vc-data-model/#credentials | W3C Verifiable Credential}
+ *
+ * @beta This API is likely to change without a BREAKING CHANGE notice
+ */
+ export interface IVerifyCredentialEIP712Args {
```

Comment:
Since this is identical to the arguments from `ICredentialIssuer`, maybe we can remove them?

Reply:
I just realized that `credential-w3c` is not a dependency here.. feel free to ignore my previous note.